ReText/window.py

```diff
@@ -943,8 +944,40 @@ def standardPrinter(self, title):
 		printer = QPrinter(QPrinter.HighResolution)
 		printer.setDocName(title)
 		printer.setCreator('ReText %s' % app_version)
+		if globalSettings.paperSize:
+			pageSize = self.getPageSizeByName(globalSettings.paperSize)
+			if pageSize is not None:
+				printer.setPaperSize(pageSize)
+			else:
+				QMessageBox.warning(self, '',
+					self.tr('Unrecognized PaperSize setting "%s".' %
+					globalSettings.paperSize))
 		return printer
 
+	def getPageSizeByName(self, pageSizeName):
+		""" Returns a validated PageSize instance
+		corresponding to the given name. Returns
+		None if the name is not a valid PageSize.
+		"""
+
+		pageSize = None
+		match = None
+		if pageSizeName:
+			match = list(filter(lambda y:y.lower()==pageSizeName.lower(),
+						self.availablePageSizes()))
+		if match:
+			pageSize = getattr(QPagedPaintDevice, match[0])
+		return pageSize
```

Comment:
Can you please still use a dictionary? Then this function will be just 3 lines instead of current 8.

```python
lowerCaseNames = {pageSize.lower(): pageSize for pageSize in self.availablePageSizes()}
if pageSizeName.lower() in lowerCaseNames:
    return getattr(QPagedPaintDevice, lowerCaseNames[pageSizeName])
```

(And it will still return None if there is no match.)

Reply:
Done.